components/sync_manager/ios/SyncManager/SyncManagerTelemetry.swift

```diff
@@ -0,0 +1,358 @@
+    GleanMetrics.CreditcardsSync.finishedAt.set(base.finishedAt)
+
+    if base.applied > 0 {
+        GleanMetrics.CreditcardsSync.incoming["applied"].add(base.applied)
+    }
+
+    if base.failedToApply > 0 {
+        GleanMetrics.CreditcardsSync.incoming["failed_to_apply"].add(base.failedToApply)
+    }
+
+    if base.reconciled > 0 {
+        GleanMetrics.CreditcardsSync.incoming["reconciled"].add(base.reconciled)
+    }
+
+    if base.uploaded > 0 {
+        GleanMetrics.CreditcardsSync.outgoing["uploaded"].add(base.uploaded)
+    }
+
+    if base.failedToUpload > 0 {
+        GleanMetrics.CreditcardsSync.outgoing["failed_to_upload"].add(base.failedToUpload)
+    }
+
+    if base.outgoingBatches > 0 {
+        GleanMetrics.CreditcardsSync.outgoingBatches.add(base.outgoingBatches)
+    }
+
+    if let reason = base.failureReason {
+        recordFailureReason(reason: reason,
+                            failureReasonMetric: GleanMetrics.CreditcardsSync.failureReason)
+    }
+}
+
+private func individualTabsSync(hashedFxaUid: String, engineInfo: EngineInfo) throws {
+    guard engineInfo.name == SupportedEngines.Tabs.rawValue else {
+        let message = "Expected 'tabs', got \(engineInfo.name)"
+        throw TelemetryReportingError.InvalidEngine(message: message)
+    }
+
+    let base = BaseGleanSyncPing.fromEngineInfo(uid: hashedFxaUid, info: engineInfo)
+    GleanMetrics.TabsSync.uid.set(base.uid)
+    GleanMetrics.TabsSync.startedAt.set(base.startedAt)
+    GleanMetrics.TabsSync.finishedAt.set(base.finishedAt)
+
+    if base.applied > 0 {
+        GleanMetrics.TabsSync.incoming["applied"].add(base.applied)
+    }
+
+    if base.failedToApply > 0 {
+        GleanMetrics.TabsSync.incoming["failed_to_apply"].add(base.failedToApply)
+    }
+
+    if base.reconciled > 0 {
+        GleanMetrics.TabsSync.incoming["reconciled"].add(base.reconciled)
+    }
+
+    if base.uploaded > 0 {
+        GleanMetrics.TabsSync.outgoing["uploaded"].add(base.uploaded)
+    }
+
+    if base.failedToUpload > 0 {
+        GleanMetrics.TabsSync.outgoing["failed_to_upload"].add(base.failedToUpload)
+    }
+
+    if base.outgoingBatches > 0 {
+        GleanMetrics.TabsSync.outgoingBatches.add(base.outgoingBatches)
+    }
+
+    if let reason = base.failureReason {
+        recordFailureReason(reason: reason,
+                            failureReasonMetric: GleanMetrics.TabsSync.failureReason)
+    }
+}
+
+private func recordFailureReason(reason: FailureReason,
+                                 failureReasonMetric: LabeledMetricType<StringMetricType>)
+{
+    let metric: StringMetricType? = {
+        switch reason.name {
+        case .other, .unknown:
+            return failureReasonMetric["other"]
+        case .unexpected, .http:
+            return failureReasonMetric["unexpected"]
+        case .auth:
+            return failureReasonMetric["auth"]
+        case .shutdown:
+            return nil
```

Comment:
I understand those are the ones we had in android, but are those still the best ones now? We might need a data review for changes but now could be the time to modify those while we are at it?

Or are we planning any changes for later, and this brings parity?

Reply:
What Sammy and I talked about was waiting to make improvements after the android reporting was moved into a-s so it'd be easier to make the improvements for both platforms.

Reply:
Yeah the general (optimistic) outline will be:
1. Remove iOS temp metrics
2. Bring telem parity with android/ios (<-- this PR)
3. Consolidate/Move both into our repo
4. Improve any metrics (for both platforms) + DR